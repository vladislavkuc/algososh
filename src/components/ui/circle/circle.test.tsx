import renderer from 'react-test-renderer';

const { Circle } = jest.requireActual('./circle');
const { ElementStates } = jest.requireActual('../../../types/element-states');

it('Круг с текстом отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc'/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг без текста отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг с текстом и head отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' head={'abc'}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг с текстом и элементом в head отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' head={<Circle letter='head'/>}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг с текстом и tail отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' tail={'abc'}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг с текстом и элементом в tail отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' tail={<Circle letter='tail'/>}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг с текстом и index отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' index={53}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг с текстом и пропсом isSmall отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' isSmall={true}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг в состоянии default отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' state={ElementStates.Default}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг в состоянии changing отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' state={ElementStates.Changing}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Круг в состоянии modified отрисовывается корректно', () => {
  const tree = renderer
    .create(<Circle letter='abc' state={ElementStates.Modified}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

export {};
