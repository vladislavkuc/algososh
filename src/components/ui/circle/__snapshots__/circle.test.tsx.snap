// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Круг без текста отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг в состоянии changing отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг в состоянии default отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг в состоянии modified отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг с текстом и head отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    abc
  </div>
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг с текстом и index отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  >
    53
  </p>
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг с текстом и tail отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    abc
  </div>
</div>
`;

exports[`Круг с текстом и пропсом isSmall отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined  undefined undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг с текстом и элементом в head отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined   undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        >
          head
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Круг с текстом и элементом в tail отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  >
    <div
      className="undefined "
    >
      <div
        className="text text_type_input text_color_input mb-4 undefined undefined undefined"
      />
      <div
        className="undefined   undefined"
      >
        <p
          className="text text_type_circle text_color_input undefined"
        >
          tail
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 undefined undefined"
      />
      <div
        className="text text_type_input text_color_input mt-4 undefined undefined undefined"
      />
    </div>
  </div>
</div>
`;

exports[`Круг с текстом отрисовывается корректно 1`] = `
<div
  className="undefined "
>
  <div
    className="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    className="undefined   undefined"
  >
    <p
      className="text text_type_circle text_color_input undefined"
    >
      abc
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    className="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;
